finetuning/generalists/training/histopathology/obtain_hp_datasets.py

```diff
@@ -0,0 +1,150 @@
+import os
+import numpy as np
+from math import ceil, floor
+from typing import Optional, List
+
+import torch
+import torch.utils.data as data_util
+
+import torch_em
+from torch_em.data import datasets, MinInstanceSampler, ConcatDataset
+
+
+"""NOTE: test sets for in-domain histopathology evaluation
+    - monuseg test split
+    - monusac test split
+    - bcss test samples (split intrinsically - in the new PR)
+
+length of individual loaders: @all (3 channel input images)
+    - lizard:  train - 359*2;  val - 179
+    - bcss:    train - 54*2;   val - 28
+    - monuseg: train - 15*2;   val - 7
+    - monusac: train - 84*2;   val - 41
+    - pannuke: train - 647*2;  val - 680
+"""
+
+
+def _get_train_val_split(ds, val_fraction: float = 0.2):
+    generator = torch.Generator().manual_seed(42)
+    train_ds, val_ds = data_util.random_split(ds, [1 - val_fraction, val_fraction], generator=generator)
+    return train_ds, val_ds
+
+
+class BCSSLabelTrafo:
+    def __init__(self, label_choices: Optional[List[int]] = None):
+        self.label_choices = label_choices
+
+    def __call__(self, labels: np.ndarray) -> np.ndarray:
+        """Returns the transformed bcss data labels (use-case for SAM)"""
```

Comment:
I think we should use connected components instead of simple relabeling here. This way disconnected pieces of the same semantic class would actually be split.

Reply:
Currently `label_padding_trafo` uses `label_consecutive_trafo` inside, you can add a switch to instead use `skimage.measure.label`.

Reply:
Yes, that makes sense. Thanks Constantin! (fixed it just now in the latest commit)

Reply:
Now this is a bit confusing because one of the code paths applies padding and the other one doesn't...